processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -900,6 +927,27 @@ public Rowboat apply(@Nullable Rowboat input)
         )
     );
 
+    if (segmentMetadata != null && !segmentMetadata.isEmpty()) {
+      File metadataFile = new File(v8OutDir, "metadata.drd");
+      FileOutputStream metadataFileOutputStream = null;
+      FileChannel metadataFilechannel = null;
+      try {
+        metadataFileOutputStream = new FileOutputStream(metadataFile);
+        metadataFilechannel = metadataFileOutputStream.getChannel();
+        metadataFilechannel.write(ByteBuffer.wrap(mapper.writeValueAsBytes(segmentMetadata)));
+      }
+      finally {
+        CloseQuietly.close(metadataFilechannel);
```

Comment:
fwiw I agree both that the Quietlies are probably bad and that they should be looked at in a separate PR. So I think for this PR, I wouldn't change it.